gather_metrics.py

```diff
@@ -0,0 +1,114 @@
+import socket
+import sys
+import getopt
+from datetime import datetime
+import subprocess
+
+import conf_collectd_plugin as conf
+import install_utils as utils
+import config
+
+# Python required base version, haven't tested 3 yet.
+REQ_VERSION = (2, 7)
+
+
+def check_version():
+    cur_version = sys.version_info
+    utils.print_step("  Checking python version")
+    if cur_version < REQ_VERSION:
+        sys.stderr.write(
+            "Your Python interpreter is older " +
+            "than what we have tested, we suggest upgrading " +
+            "to a newer 2.7 version before continuing.\n")
+        sys.exit()
+    utils.print_success()
+
+
+def port_scan(host, port):
+    """
+    Using ipv4, TCP connection to test whether a port is
+    being used.
+    """
+    # AF_INET specifies ipv4, SOCK_STREAM for TCP
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+    try:
+        sock.connect((host, port))
+    except socket.error:
+        return False
+    except KeyboardInterrupt:
+        print "Scanning interrupted"
+        sys.exit()
+    except socket.gaierror:
+        print "Hostname could not be resolved"
+        sys.exit()
+    else:
+        return port
+    finally:
+        sock.close()
+
+
+def detect_used_ports():
```

Comment:
Originally, I wrote this to test the tcpconns collectd plugin.  Due to the
change of direction in the project, I have moved on to writing apache
plugin.

tcpconns collected plugin has a option that detects all listening port.  If
we only care about listening port, then tcpconns can be packaged in the
default collectd configuration.
However, it also has the capability to collect data that user sends to a
remote server.  To do so, the plugin requires the user to specify such
port.  I was planning on use this to detect such port, but detection can
take a while.